sklearn/discriminant_analysis.py

```diff
@@ -397,7 +397,8 @@ def _solve_svd(self, X, y):
         # (n_classes) centers
         _, S, V = linalg.svd(X, full_matrices=0)
 
-        self.explained_variance_ratio_ = S[:self.n_components] / S.sum()
+        self.explained_variance_ratio_ = (S**2 / np.sum(
+            S**2))[:self.n_components]
```

Comment:
Add 4 more spaces here to conform to PEP 8 indentation.
https://www.python.org/dev/peps/pep-0008/#indentation